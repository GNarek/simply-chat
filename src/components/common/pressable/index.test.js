import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Pressable from './index';

test('Pressable component renders correctly', () => {
  const tree = renderer.create(<Pressable />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Pressable component takes correct hitSlop', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Pressable hitSlop={{
        top: 20,
        bottom: 20,
        left: 20,
        right: 20,
      }}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({
      hitSlop: {
        top: 12,
        bottom: 19,
      },
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
