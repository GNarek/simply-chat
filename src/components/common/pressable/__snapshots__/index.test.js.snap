// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pressable component renders correctly 1`] = `
<View
  accessible={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 5,
      "right": 5,
      "top": 10,
    }
  }
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;

exports[`Pressable component takes correct hitSlop renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  style={Object {}}
/>
`;

exports[`Pressable component takes correct hitSlop renders as expected 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
  hitSlop={
    Object {
      "bottom": 19,
      "top": 12,
    }
  }
  style={Object {}}
/>
`;
